spikeinterface/toolkit/qualitymetrics/misc_metrics.py

```diff
@@ -139,23 +151,28 @@ def compute_isi_violations(waveform_extractor, isi_threshold_ms=1.5):
     seg_durations = [recording.get_num_samples(i) / fs for i in range(num_segs)]
     total_duraion = np.sum(seg_durations)
 
-    isi_threshold = (isi_threshold_ms / 1000. * fs)
+    isi_threshold_s = isi_threshold_ms / 1000
+    min_isi_s = min_isi_ms / 1000
 
-    isi_violations_count = {}
     isi_violations_rate = {}
 
+    # all units converted to seconds
     for unit_id in unit_ids:
         num_violations = 0
+        num_spikes = 0
         for segment_index in range(num_segs):
-            st = sorting.get_unit_spike_train(unit_id=unit_id, segment_index=segment_index)
-            isi = np.diff(st)
-            num_violations += np.sum(isi < isi_threshold)
+            spike_train_s = sorting.get_unit_spike_train(unit_id=unit_id, segment_index=segment_index) * fs
```

Comment:
You should work at sample level, this avoid a vector multiplication and more memory

Reply:
I think it's not too bad

Reply:
yes but it was better before. One division is better that multiply a several full vector...